react/features/lobby/components/LobbyModeButton.js

```diff
@@ -0,0 +1,77 @@
+// @flow
+
+import { IconMeetingUnlocked, IconMeetingLocked } from '../../base/icons';
+import { isLocalParticipantModerator } from '../../base/participants';
+import AbstractButton, { type Props as AbstractProps } from '../../base/toolbox/components/AbstractButton';
+import { translate } from '../../base/i18n';
+import { connect } from '../../base/redux';
+
+import { setLobbyModeEnabled, showEnableLobbyModeDialog } from '../actions';
+
+type Props = AbstractProps & {
+
+    /**
+     * The Redux Dispatch function.
+     */
+    dispatch: Function,
+
+    /**
+     * True if the lobby mode is currently enabled for this conference.
+     */
+    lobbyModeEnabled: boolean
+};
+
+/**
+ * Component to render the lobby mode initiator button.
+ */
+class LobbyModeButton extends AbstractButton<Props, any> {
+    accessibilityLabel = 'toolbar.accessibilityLabel.lobbyButton';
+    icon = IconMeetingUnlocked;
+    label = 'toolbar.lobbyButtonEnable';
+    toggledLabel = 'toolbar.lobbyButtonDisable'
+    toggledIcon = IconMeetingLocked;
+
+    /**
+     * Callback for the click event of the button.
+     *
+     * @returns {void}
+     */
+    _handleClick() {
+        const { dispatch } = this.props;
+
+        if (this._isToggled()) {
+            dispatch(setLobbyModeEnabled(false));
+        } else {
+            dispatch(showEnableLobbyModeDialog());
+        }
+    }
+
+    /**
+     * Function to define the button state.
+     *
+     * @returns {boolean}
+     */
+    _isToggled() {
+        return this.props.lobbyModeEnabled;
+    }
+}
+
+/**
+ * Maps part of the Redux store to the props of this component.
+ *
+ * @param {Object} state - The Redux state.
+ * @param {Props} ownProps - The own props of the component.
+ * @returns {Props}
+ */
+export function _mapStateToProps(state: Object): $Shape<Props> {
+    const { conference } = state['features/base/conference'];
+    const lobbySupported = conference && conference.isLobbySupported();
+    const { membersOnlyEnabled } = conference?.room || {};
```

Comment:
isn't there a getter for this? I'm not ure it's kosher to peek into conference.room...

Reply:
I checked LJM and couldn't find a getter

Reply:
Let' add it then. @damencho

Reply:
Why do we need that? We should not access that internal room I think. I can add isMemebersOnly in JitsiConference?